### ISSUE TITLE
Deleting iRules from Virtual Server Configs

### ISSUE THREAD

**Issue by atyanh** [F5Networks/terraform-provider-bigip#923]
### Environment

 * TMOS/Bigip Version: BIG-IP 16.1.2.2 Build 0.0.28 Point Release 2
 * Terraform Version:  v1.6.6
 * Terraform bigip provider Version:  v1.20.2

### Summary

Terraform doesn't change iRule list in Virtual Server configurations when I leave the list empty. It works correctly if I leave at least one iRule but if we delete all and leave it the array empty it doesn't do anything.

### Steps To Reproduce

For example 
 `    irules       = [ bigip_ltm_irule.rule1.name, bigip_ltm_irule.rule2.name ]  `
 This works correctly, but when I delete iRules it  shows on the plan but doesn't do anything
`      irules       = [ ] `

**Comment by pgouband** [comment]
Hi @atyanh,

I tested and it works uring:
irules = [ "" ]

Could you test?

**Comment by atyanh** [comment]
> Hi @atyanh,
> 
> I tested and it works uring: irules = [ "" ]
> 
> Could you test?


You are right it worked, thank you.

But now every time I execute terraform plan it shows that the rules will be updated.

```
  ~ resource "bigip_ltm_virtual_server" "test-proj_https" {
        id                         = "/Common/https-app-tf-12347"
      ~ irules                     = [
          + "",
        ]
        name                       = "/Common/https-app-tf-12347"
        # (19 unchanged attributes hidden)
    }
```

Is this okay? I think it will interfere with work when I provision some resources.

**Comment by pgouband** [comment]
Hi @atyanh,

You're right about the output of plan but you can avoid that removing the line "irules".

**Comment by atyanh** [comment]
Thank you for your answer, but I'm creating modules and should always pass a value for irules attribute.Can I find a way to do that?

**Comment by pgouband** [comment]
Hi @atyanh,

Can you create two modules? One with irules list and one without?

**Comment by atyanh** [comment]
> Hi @atyanh,
> 
> Can you create two modules? One with irules list and one without?

Yes I can, but I don't think it's the best solution to duplicate 4 modules just for irules. I thought  it would be easy to resolve and would like to help with developing the provider.Whatever thank you :)

**Comment by pgouband** [comment]
Hi,

Thanks for reporting. Added to the backlog and internal tracking ID for this request is: INFRAANO-1417.

**Comment by atyanh** [comment]
Okay Thank you

**Comment by RavinderReddyF5** [comment]
fixed in v1.21.0
